airbyte-tests/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java

```diff
@@ -0,0 +1,276 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.test.airbyte_test_container;
+
+import com.google.api.client.util.Preconditions;
+import com.google.common.collect.Maps;
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.HealthApi;
+import io.airbyte.api.client.invoker.ApiClient;
+import io.airbyte.api.client.invoker.ApiException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Scanner;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.SocatContainer;
+import org.testcontainers.containers.output.OutputFrame;
+
+/**
+ * The goal of this class is to make it easy to run the Airbyte docker-compose configuration from
+ * test containers. This helps me it easy to stop the test container without deleting the volumes
+ * { @link AirbyteTestContainer#stopRetainVolumes() }. It waits for Airbyte to be ready. It also
+ * handles the nuances of configuring the Airbyte docker-compose configuration in test containers.
+ */
+public class AirbyteTestContainer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AirbyteTestContainer.class);
+
+  private final File dockerComposeFile;
+  private final Map<String, String> env;
+  private final Map<String, Consumer<String>> customServiceLogListeners;
+
+  private DockerComposeContainer<?> dockerComposeContainer;
+
+  public AirbyteTestContainer(final File dockerComposeFile,
+                              final Map<String, String> env,
+                              final Map<String, Consumer<String>> customServiceLogListeners) {
+    this.dockerComposeFile = dockerComposeFile;
+    this.env = env;
+    this.customServiceLogListeners = customServiceLogListeners;
+  }
+
+  /**
+   * Starts Airbyte docker-compose configuration. Will block until the server is reachable or it times
+   * outs.
+   */
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public void start() throws IOException, InterruptedException {
+    final File cleanedDockerComposeFile = prepareDockerComposeFile(dockerComposeFile);
+    dockerComposeContainer = new DockerComposeContainer(cleanedDockerComposeFile).withEnv(env);
```

Comment:
that's pretty awesome!

Reply:
yeah. credit goes to @subodh1810 for figuring this out originally. i'm just putting a new coat of paint on it.